### DISCUSSION TITLE
"Cannot find a handler for Task type" in playbooks

### DISCUSSION THREAD

**Discussion by nlegaillart-dri** [choria-io/general#45] [Q&A] [answered]
Hi all,

We are having some trouble to make playbooks work. However we are able to launch tasks using `mco` and `colt` commands.

For instance the `apt` Puppet module tasks work just fine using command line (`mco tasks run apt --action=update -I <Host>`), but the following playbook code:

```puppet
plan playbooks::apt_update (
  Array[String[1]] $hosts
) {
  choria::task("apt",
    "action" => "update",
    "nodes"  => $hosts,
  )
}
```

fails with the following error:

```
Error: Evaluation Error: Error while evaluating a Function Call, Cannot find a handler for Task type apt (file: ...)
```

This "Cannot find a handler" error is encountered on every `choria::tasks` we tried (at least, `apt` and `exec` tasks)

Is there something we are missing? It is our first try at playbooks so it could clearly be the case ;-)

Thanks in advance for your help,

--
Nicolas

**Comment by ripienaar** [comment, marked as answer]
I am guessing you want to run this kind of task? A puppet task? There's an unfortunate collision of terminology but seems like what you want https://choria.io/docs/tasks/usage/#running-a-task-1

**Reply by nlegaillart-dri** [reply]
Hi R.I,

Yes, this is exactly what we want to achieve. Thank you for the kick answer, and the good pointer: we were not using the right syntax (instead of calling `choria::task` directly, we have to call `choria::run_playbook("choria::tasks::run"...`)

Sorry for this rookie question, I think we can close this discussion now!
